sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpInputStreamRequestBody.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.okhttp.implementation;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.implementation.util.InputStreamContent;
+import com.azure.core.util.logging.ClientLogger;
+import okhttp3.MediaType;
+import okio.BufferedSink;
+import okio.Okio;
+import okio.Source;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * An {@link okhttp3.RequestBody} subtype that sends {@link InputStreamContent} in an unbuffered manner.
+ */
+public class OkHttpInputStreamRequestBody extends OkHttpStreamableRequestBody<InputStreamContent> {
+
+    private static final ClientLogger LOGGER = new ClientLogger(OkHttpInputStreamRequestBody.class);
+
+    private final AtomicBoolean bodySent = new AtomicBoolean(false);
+
+    public OkHttpInputStreamRequestBody(InputStreamContent content, HttpHeaders httpHeaders, MediaType mediaType) {
+        super(content, httpHeaders, mediaType);
+    }
+
+    @Override
+    public void writeTo(BufferedSink bufferedSink) throws IOException {
+        if (bodySent.compareAndSet(false, true)) {
+            Source source = Okio.source(content.toStream());
+            bufferedSink.writeAll(source);
```

Comment:
What happens if the body is >2GB here?

Reply:
This thing works similarly to BufferedInputStream/BufferedOutputStream. i.e. it doesn't buffer the whole thing.

![image](https://user-images.githubusercontent.com/61715331/168652740-e93346cd-a61c-4bd2-bbf2-f247b7795d0a.png)